languages/go/sample-basics.go

```diff
@@ -0,0 +1,84 @@
+
+package main
+
+import "fmt"
+
+func main() {
+  fmt.Println("+++ Sample Basics in Go +++")
+
+  println("Generating sample slice of ints...")
+  sample_numbers := []int{1, 2, 3, 4, 5, 10, 100}
+  println(stringify(sample_numbers))
+
+  println("Generating sample slice of strings...")
+  sample_strings := []string{"abc", "123", "xyz"}
+  println(stringify(sample_strings))
+
+  println("Summing sample ints...")
+  sample_numbers_sum := sum(sample_numbers...)
+  println(stringify(sample_numbers_sum))
+
+  println("Concatenating sample strings...")
+  sample_strings_concat := concat(sample_strings...)
+  println(stringify(sample_strings_concat))
+
+  println("Building and running sample tests...")
+  type Test struct {
+    name string
+    expected interface{}
+    actual interface{}
+  }
+  tests := []Test{
+    Test{name: "sample_numbers_sum", expected: 125, actual: sample_numbers_sum},
+    Test{name: "sample_string_concat", expected: "abc123xyz", actual: sample_strings_concat},
```

Comment:
Let's make these test name fields better. I was originally exploring reflection and lambdas to see if we could target a property/variable name with a specific anonymous test method just for fun, but I removed it.